### PR TITLE
docs: misc

### PR DESCRIPTION
**Pull request by aalemayhu**
A couple of changes to recent issues. Please see individual changes for details.

### CODE REVIEW

**Review by aanm — APPROVED** [review]
This is great! Just to be sure, we do have a unit / runtime test for every policy in `examples/policy`, right?

File: Documentation/gettingstarted.rst [hunk 1]
```diff
@@ -701,19 +701,7 @@ of the same TCP/UDP connection.
...
-        "ingress": [{
-            "fromEndpoints": [
-                {"matchLabels":{"id":"app2"}}
-            ],
-            "toPorts": [{
-                    "ports": [{"port": "80", "protocol": "TCP"}]
-            }]
-        }]
-    }]
+.. literalinclude:: ../examples/policies/cilium_dkr_demo_l3-l4-policy-170817.json
```
Comment (tgraf) [inline Documentation/gettingstarted.rst]: Can you move the files into `examples/policies/getting-started/`?

Any reason for the date in the filename?

File: Documentation/policy.rst [hunk 2]
```diff
@@ -151,55 +151,27 @@ Example (Basic)
...
 
-        [{
-            "endpointSelector": {"matchLabels": {"role":"backend"}},
-            "ingress": [{
-                "fromEndpoints": [
-                  {"matchLabels":{"role":"frontend"}}
-                ]
-            }]
-        }]
+.. literalinclude:: ../examples/policies/basic.json
```
Comment (tgraf) [inline Documentation/policy.rst]: `examples/policies/l3.json/`

File: Documentation/policy.rst [hunk 3]
```diff
@@ -151,55 +151,27 @@ Example (Basic)
...
-            "endpointSelector": {"matchLabels": {"role":"backend"}},
-            "ingress": [{
-                "fromEndpoints": [
-                  {"matchLabels":{"role":"frontend"}}
-                ]
-            }]
-        }]
+the label ``env=prod`` in order for access to be granted:
 
+.. literalinclude:: ../examples/policies/basic_requires.json
```
Comment (tgraf) [inline Documentation/policy.rst]: `requires.json`

File: Documentation/policy.rst [hunk 4]
```diff
@@ -462,35 +397,18 @@ Example (L4)
...
-                  {"matchLabels":{"role":"frontend"}}
-                ],
-                "toPorts": [
-                    {"ports":[ {"port": "80", "protocol": "TCP"}]}
-                ]
-            }]
-        }]
+communicate using using TCP on port 80:
+
+.. literalinclude:: ../examples/policies/l4_labels.json
```
Comment (tgraf) [inline Documentation/policy.rst]: `l3_l4_combined.json`

File: Documentation/policy.rst [hunk 5]
```diff
@@ -499,20 +417,9 @@ This example is similar to the previous, but rather than restricting
...
-                ]
-              }, {
-                "toPorts": [
-                    {"ports":[ {"port": "80", "protocol": "TCP"}]}
-                ]
-            }]
-        }]
+endpoint that is communicating over TCP on port 80:
+
+.. literalinclude:: ../examples/policies/l4_labels_all.json
```
Comment (tgraf) [inline Documentation/policy.rst]: `multi_rule.json`

**Review by tgraf — CHANGES_REQUESTED** [review]

**Comment by aalemayhu** [comment]
@anam I don't think these are in the CI, but there are similar policies there. Did not add any new test here yet, was thinking to first get the location in the tree. Then later discuss how to do the automated validation. Should be fairly straightforward for the JSON case since we just need to loop over a list of files and run the validation. Not sure about the YAML case, how is validation performed there?

**Comment by aanm** [comment]
>@anam I don't think these are in the CI, but there are similar policies there. Did not add any new test here yet, was thinking to first get the location in the tree. Then later discuss how to do the automated validation. Should be fairly straightforward for the JSON case since we just need to loop over a list of files and run the validation. Not sure about the YAML case, how is validation performed there?

Aren't the YAML examples used in k8s? That would be validated by k8s itself. The creation of tests for these examples can be done later.

File: Documentation/gettingstarted.rst (hunk 1 above)
Comment (aalemayhu) [reply Documentation/gettingstarted.rst]: No, not really, was using unique nearby string I saw. Can change it something else, just let me know what? Moving it into the new directory `examples/policies/getting-started/`.

File: Documentation/policy.rst (hunk 2 above)
Comment (aalemayhu) [reply Documentation/policy.rst]: Fixed

File: Documentation/policy.rst (hunk 3 above)
Comment (aalemayhu) [reply Documentation/policy.rst]: Fixed

File: Documentation/policy.rst (hunk 4 above)
Comment (aalemayhu) [reply Documentation/policy.rst]: Fixed

File: Documentation/policy.rst (hunk 5 above)
Comment (aalemayhu) [reply Documentation/policy.rst]: Fixed

**Review by ianvernon — APPROVED** [review]
This looks great! The use of `literalincludes` will help ensure our documentation examples never fall out of step with what Cilium accepts. I second what @aanm said in the need for a runtime test for the JSON policy examples. Please create a separate GitHub issue for that.

**Review by tgraf — APPROVED** [review]

File: tests/00-check-examples.sh [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+
+TEST_NAME=$(get_filename_without_extension $0)
+LOGS_DIR="${dir}/cilium-files/${TEST_NAME}/logs"
+redirect_debug_logs ${LOGS_DIR}
+
+set -ex
+
+for p in ../examples/policies/*.json ../examples/demo/*.json ; do
+ if ! cilium policy validate $p; then
```
Comment (aanm) [inline tests/00-check-examples.sh:15]: This won't be enough, please create a GH issue to track the enforcement of the policy, not only the validation.

**Review by aanm — APPROVED** [review]

File: tests/00-check-examples.sh (hunk 6 above)
Comment (aalemayhu) [reply tests/00-check-examples.sh:15]: Created #1912, please add to it if something is missing.

**Comment by aalemayhu** [comment]
Latest failure is in the k8s tests and looks unrelated.
https://jenkins.cilium.io/job/cilium/job/cilium/job/PR-1887/8/execution/node/29/log/

**Pull request merged**